session.go

```diff
@@ -308,8 +308,15 @@ func (s *Session) processIncomingMessage(msg messages.Message) error {
 	return nil
 }
 
+func (s *Session) Connected() bool {
+	return s.goodBye == nil
+}
+
 func (s *Session) Register(procedure string, handler InvocationHandler,
 	options map[string]any) (*Registration, error) {
+	if !s.Connected() {
+		return nil, fmt.Errorf("session not connected")
```

Comment:
the error message should be tailered for each type of WAMP actions.
```suggestion
		return nil, fmt.Errorf("cannot register procedure: session not established")
```